pkg/gadgets/tcptracer/tracer/standard/tracer.go

```diff
@@ -67,11 +76,14 @@ func NewTracer(config *tracer.Config, resolver containercollection.ContainerReso
 		eventCallback:      eventCallback,
 		resolver:           resolver, // not used right now but could be useful in the future
 		node:               node,
+		mountNsMapPinPath:  mountNsMapPath,
 	}, nil
 }
 
 func (t *Tracer) Stop() {
 	if err := t.StandardTracerBase.Stop(); err != nil {
 		t.eventCallback(types.Base(eventtypes.Warn(err.Error(), t.node)))
 	}
+
+	os.Remove(t.mountNsMapPinPath)
```

Comment:
All bcc-based gadgets have this code pattern. We could have the `Tracer struct` embed a "parent" struct with some helpers to pin and unpin the map. Idea for a future PR.

Reply:
We already have the "parent" struct you mentioned: `StandardTracerBase`, I modified the code to handle this logic there and it seems much better now.